### PR TITLE
Add gcp_conn_id argument to GoogleDriveToLocalOperator

### PR DESCRIPTION
**Pull request by aspain**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by aspain** [comment]
For https://github.com/apache/airflow/issues/24494

**Review by turbaszek — APPROVED** [review]

**Comment by mik-laj** [comment]
Black is still sad 😿

File: tests/providers/google/cloud/transfers/test_gdrive_to_local.py [hunk 1]
```diff
@@ -18,28 +18,36 @@
 from tempfile import NamedTemporaryFile
 from unittest import TestCase, mock
 
-from airflow.providers.google.cloud.transfers.gdrive_to_local import GoogleDriveToLocalOperator
+from airflow.providers.google.cloud.transfers.gdrive_to_local import (
+    GoogleDriveToLocalOperator,
+)
```
Comment (uranusjr) [inline tests/providers/google/cloud/transfers/test_gdrive_to_local.py]:
```suggestion
from airflow.providers.google.cloud.transfers.gdrive_to_local import GoogleDriveToLocalOperator
```

File: tests/providers/google/cloud/transfers/test_gdrive_to_local.py [hunk 2]
```diff
@@ -18,28 +18,36 @@
...
 FOLDER_ID = "1234567890qwerty"
 FILE_NAME = "file.pdf"
+GCP_CONN_ID = "google_cloud_default"
 
 
 class TestGoogleDriveToLocalOperator(TestCase):
-    @mock.patch("airflow.providers.google.cloud.transfers.gdrive_to_local.GoogleDriveHook")
+    @mock.patch(
+        "airflow.providers.google.cloud.transfers.gdrive_to_local.GoogleDriveHook"
+    )
```
Comment (uranusjr) [inline tests/providers/google/cloud/transfers/test_gdrive_to_local.py]:
```suggestion
    @mock.patch("airflow.providers.google.cloud.transfers.gdrive_to_local.GoogleDriveHook")
```

**Review by uranusjr — DISMISSED** [review]

**Pull request merged**